numba/core/cpu.py

```diff
@@ -205,7 +205,7 @@ def create_cfunc_wrapper(self, library, fndesc, env, call_helper):
 
         status, out = self.call_conv.call_function(
             builder, wrapper_callee, fndesc.restype, fndesc.argtypes,
-            wrapfn.args, attrs=('noinline',))
+            wrapfn.args)
```

Comment:
After discussion with @stuartarchibald, this will need to be undone or it will leads to many optimization issues.

The alternative is to make first-class function use the Numba function without the wrapper when possible.

Reply:
Yeah, I think so. making it always inline could lead to many issues. One potential drawback could increase the compilation time, if the body of `FunctionType` is huge. Or maybe it's a bug in inline-related passes...